lib/agent.js

```diff
@@ -22,6 +23,18 @@ function defaultFactory (origin, opts) {
     : new Pool(origin, opts)
 }
 
+function getDNSResolver (options) {
+  if (options.DNSResolver instanceof DNSResolver) {
```

Comment:
I think it's okay to assume that if this option is present, it's an instance of DNSResolver

Reply:
But we allow custom `DNSResolver` instances that do not necessarily extend from `DNSResolver`; maybe we should check if the required methods are provided and fail if not?

Reply:
The option is not always present, it allows people to set a custom DNSResolver instead of our one. They could inject another module or develop their own.

I added the check on the function needed.